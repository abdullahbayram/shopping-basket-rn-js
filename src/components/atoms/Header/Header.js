import * as React from 'react';
import PropTypes from 'prop-types';
import { Appbar } from 'react-native-paper';

const Header = ({ handleSearch = null, handleMore = null, onBackPress = null, title }) => {
  return (
    <Appbar.Header>
      {onBackPress && <Appbar.BackAction onPress={onBackPress} />}
      <Appbar.Content title={title} />
      {handleSearch && <Appbar.Action icon="magnify" onPress={handleSearch} />}
      {handleMore && <Appbar.Action icon="dots-vertical" onPress={handleMore} />}
    </Appbar.Header>
  );
};

Header.propTypes = {
  handleSearch: PropTypes.func,
  handleMore: PropTypes.func,
  onBackPress: PropTypes.func,
  title: PropTypes.string.isRequired,
};

export default Header;
