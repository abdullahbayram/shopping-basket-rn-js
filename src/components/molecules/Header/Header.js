import * as React from 'react';
import PropTypes from 'prop-types';
import Appbar from '../../atoms/Appbar';
import accessibilityLabels from '../../../constants/accessibilityLabels';

const Header = ({ handleSearch = null, handleMore = null, onBackPress = null, title }) => {
  return (
    <Appbar.Header>
      {onBackPress && <Appbar.BackAction accessibilityRole={accessibilityLabels.back} onPress={onBackPress} />}
      <Appbar.Content title={title} />
      {handleSearch && (
        <Appbar.Action accessibilityRole={accessibilityLabels.search} icon="magnify" onPress={handleSearch} />
      )}
      {handleMore && (
        <Appbar.Action accessibilityRole={accessibilityLabels.more} icon="dots-vertical" onPress={handleMore} />
      )}
    </Appbar.Header>
  );
};

Header.propTypes = {
  handleSearch: PropTypes.func,
  handleMore: PropTypes.func,
  onBackPress: PropTypes.func,
  title: PropTypes.string.isRequired,
};

export default Header;
