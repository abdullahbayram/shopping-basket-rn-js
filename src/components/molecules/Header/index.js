import * as React from 'react';
import PropTypes from 'prop-types';
import { Appbar } from '../../atoms';

const Header = ({ onBackPress = null, title }) => {
  return (
    <Appbar.Header>
      {onBackPress && <Appbar.BackAction testID="back-action-button" onPress={onBackPress} />}
      <Appbar.Content title={title} />
    </Appbar.Header>
  );
};

Header.propTypes = {
  onBackPress: PropTypes.func,
  title: PropTypes.string.isRequired,
};

export default Header;
