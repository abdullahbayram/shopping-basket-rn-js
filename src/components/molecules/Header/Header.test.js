import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react-native';
import Header from '.';

describe('<Header />', () => {
  const mockOnBackPress = jest.fn();
  const mockHandleMore = jest.fn();
  const title = 'Test Title';

  afterEach(() => {
    jest.clearAllMocks();
  });

  it('renders the title correctly', () => {
    render(<Header title={title} />);

    expect(screen.getByText(title)).toBeTruthy();
  });

  it('matches the snapshot', () => {
    const { toJSON } = render(<Header title={title} handleMore={mockHandleMore} onBackPress={mockOnBackPress} />);
    expect(toJSON()).toMatchSnapshot();
  });

  it('calls onBackPress when the back button is clicked', () => {
    render(<Header title={title} onBackPress={mockOnBackPress} />);

    const backButton = screen.getByTestId('back-action-button'); // Use testID for targeting
    fireEvent.press(backButton);

    expect(mockOnBackPress).toHaveBeenCalled();
  });

  it('does not render back button when onBackPress is not provided', () => {
    render(<Header title={title} />);

    const backButton = screen.queryByTestId('back-action-button');
    expect(backButton).toBeNull();
  });
});
