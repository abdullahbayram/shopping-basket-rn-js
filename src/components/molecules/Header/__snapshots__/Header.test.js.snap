// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> matches the snapshot 1`] = `
<View
  style={
    [
      {
        "backgroundColor": "rgba(255, 251, 254, 1)",
        "elevation": 0,
        "paddingHorizontal": 0,
        "paddingTop": 0,
        "zIndex": 0,
      },
      {},
      {},
    ]
  }
  testID="appbar-header-root-layer"
>
  <View
    collapsable={false}
    style={
      {
        "backgroundColor": "rgba(255, 251, 254, 1)",
        "height": 64,
        "shadowColor": "#000",
        "shadowOffset": {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="appbar-header-outer-layer"
  >
    <View
      collapsable={false}
      style={
        {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 251, 254, 1)",
          "flex": 1,
          "flexDirection": "row",
          "paddingBottom": undefined,
          "paddingHorizontal": 4,
          "paddingLeft": undefined,
          "paddingRight": undefined,
          "paddingTop": undefined,
          "shadowColor": "#000",
          "shadowOffset": {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
        }
      }
      testID="appbar-header"
    >
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "height": 40,
            "margin": 6,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "width": 40,
          }
        }
        testID="icon-button-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 20,
              "borderWidth": 0,
              "elevation": 0,
              "flex": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="icon-button-container"
        >
          <View
            accessibilityComponentType="button"
            accessibilityLabel="Back"
            accessibilityRole="Back Button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityTraits="button"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            centered={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                [
                  {
                    "alignItems": "center",
                    "flexGrow": 1,
                    "justifyContent": "center",
                  },
                  {
                    "borderRadius": 20,
                  },
                ],
              ]
            }
            testID="icon-button"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  {
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": [
                      {
                        "rotate": "0deg",
                      },
                    ],
                  }
                }
                testID="cross-fade-icon-current"
              >
                <View
                  style={
                    [
                      {
                        "alignItems": "center",
                        "justifyContent": "center",
                      },
                      {
                        "height": 24,
                        "transform": [
                          {
                            "scaleX": 1,
                          },
                        ],
                        "width": 24,
                      },
                    ]
                  }
                >
                  <Image
                    accessibilityIgnoresInvertColors={true}
                    source={1}
                    style={
                      [
                        {
                          "resizeMode": "contain",
                        },
                        {
                          "height": 21,
                          "tintColor": "rgba(28, 27, 31, 1)",
                          "width": 21,
                        },
                      ]
                    }
                  />
                </View>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        pointerEvents="box-none"
        style={
          [
            {
              "flex": 1,
              "paddingHorizontal": 12,
            },
            {
              "paddingHorizontal": 0,
            },
            [
              false,
              {
                "alignItems": "center",
              },
              undefined,
            ],
          ]
        }
        testID="appbar-content"
      >
        <Text
          accessibilityRole="header"
          accessibilityTraits="header"
          accessible={true}
          numberOfLines={1}
          style={
            [
              {
                "textAlign": "left",
              },
              {
                "color": "rgba(28, 27, 31, 1)",
                "writingDirection": "ltr",
              },
              [
                {
                  "fontFamily": "System",
                  "fontSize": 22,
                  "fontWeight": "400",
                  "letterSpacing": 0,
                  "lineHeight": 28,
                },
                [
                  {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontSize": 22,
                    "fontWeight": "400",
                    "letterSpacing": 0,
                    "lineHeight": 28,
                  },
                  false,
                  undefined,
                ],
              ],
            ]
          }
          testID="appbar-content-title-text"
        >
          Test Title
        </Text>
      </View>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "height": 40,
            "margin": 6,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "width": 40,
          }
        }
        testID="icon-button-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 20,
              "borderWidth": 0,
              "elevation": 0,
              "flex": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="icon-button-container"
        >
          <View
            accessibilityComponentType="button"
            accessibilityRole="Search Button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityTraits="button"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            centered={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                [
                  {
                    "alignItems": "center",
                    "flexGrow": 1,
                    "justifyContent": "center",
                  },
                  {
                    "borderRadius": 20,
                  },
                ],
              ]
            }
            testID="icon-button"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  {
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": [
                      {
                        "rotate": "0deg",
                      },
                    ],
                  }
                }
                testID="cross-fade-icon-current"
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "color": "rgba(73, 69, 79, 1)",
                        "fontSize": 24,
                      },
                      [
                        {
                          "lineHeight": 24,
                          "transform": [
                            {
                              "scaleX": 1,
                            },
                          ],
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                      ],
                      {
                        "fontFamily": "material-community",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  󰍉
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
      <View
        collapsable={false}
        style={
          {
            "backgroundColor": "transparent",
            "borderRadius": 20,
            "height": 40,
            "margin": 6,
            "shadowColor": "#000",
            "shadowOffset": {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
            "width": 40,
          }
        }
        testID="icon-button-container-outer-layer"
      >
        <View
          collapsable={false}
          style={
            {
              "backgroundColor": "transparent",
              "borderColor": "rgba(121, 116, 126, 1)",
              "borderRadius": 20,
              "borderWidth": 0,
              "elevation": 0,
              "flex": 1,
              "overflow": "hidden",
              "shadowColor": "#000",
              "shadowOffset": {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
            }
          }
          testID="icon-button-container"
        >
          <View
            accessibilityComponentType="button"
            accessibilityRole="More Options Button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": false,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityTraits="button"
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            centered={true}
            collapsable={false}
            focusable={true}
            hitSlop={
              {
                "bottom": 6,
                "left": 6,
                "right": 6,
                "top": 6,
              }
            }
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "overflow": "hidden",
                },
                [
                  {
                    "alignItems": "center",
                    "flexGrow": 1,
                    "justifyContent": "center",
                  },
                  {
                    "borderRadius": 20,
                  },
                ],
              ]
            }
            testID="icon-button"
          >
            <View
              style={
                [
                  {
                    "alignItems": "center",
                    "justifyContent": "center",
                  },
                  {
                    "height": 24,
                    "width": 24,
                  },
                ]
              }
            >
              <View
                collapsable={false}
                style={
                  {
                    "bottom": 0,
                    "left": 0,
                    "opacity": 1,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                    "transform": [
                      {
                        "rotate": "0deg",
                      },
                    ],
                  }
                }
                testID="cross-fade-icon-current"
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    [
                      {
                        "color": "rgba(73, 69, 79, 1)",
                        "fontSize": 24,
                      },
                      [
                        {
                          "lineHeight": 24,
                          "transform": [
                            {
                              "scaleX": 1,
                            },
                          ],
                        },
                        {
                          "backgroundColor": "transparent",
                        },
                      ],
                      {
                        "fontFamily": "material-community",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      {},
                    ]
                  }
                >
                  󰇙
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
